projects/wallet-template/src/containers/WalletPopup/pages/Accounts/ImportAccounts.tsx

```diff
@@ -0,0 +1,349 @@
+import { ArrowLeft, Key, NotepadText, ChevronDown, Check } from "lucide-react"
+import { rpc } from "../../api"
+import { SubmitHandler, useForm, Controller } from "react-hook-form"
+import { useNavigate } from "react-router-dom"
+import { useState } from "react"
+import {
+  entropyToMiniSecret,
+  sr25519,
+  ed25519,
+  ecdsa,
+  mnemonicToEntropy,
+} from "@polkadot-labs/hdkd-helpers"
+import { fromHex, toHex } from "@polkadot-api/utils"
+import { bytesToHex } from "@noble/ciphers/utils"
+import { sr25519CreateDerive } from "@polkadot-labs/hdkd"
+import { networks } from "./networks"
+
+type FormFields = {
+  mnemonicInput?: string
+  key?: string
+  scheme: "Sr25519" | "Ed25519" | "Ecdsa"
+  networks: {
+    polkadot: boolean
+    westend: boolean
+    kusama: boolean
+  }
+  keysetName: string
+}
+
+type Tab = "private" | "mnemonic"
+
+export function ImportAccounts() {
+  const navigate = useNavigate()
+  const [activeTab, setActiveTab] = useState<Tab>("private")
+  const {
+    register,
+    handleSubmit,
+    control,
+    setValue,
+    clearErrors,
+    getValues,
+    formState: { errors },
+  } = useForm<FormFields>({
+    mode: "onSubmit",
+    reValidateMode: "onSubmit",
+    defaultValues: {
+      scheme: "Sr25519",
+      networks: {
+        polkadot: false,
+        westend: false,
+        kusama: false,
+      },
+    },
+  })
+
+  const handleChange = (chain: "kusama" | "polkadot" | "westend") => {
+    const newValue = !getValues(`networks.${chain}`)
+    return {
+      ...getValues("networks"),
+      [chain]: newValue,
+    }
+  }
+
+  const onActiveTabChanged = (tab: Tab) => {
+    setActiveTab(tab)
+    clearErrors("key")
+    setValue("key", "")
+  }
+
+  const onSubmit: SubmitHandler<FormFields> = async (data) => {
+    switch (activeTab) {
+      case "private": {
+        await rpc.client.insertKeyset({
+          _type: "PrivateKey",
+          name: data.keysetName,
+          scheme: data.scheme,
+          miniSecret: bytesToHex(entropyToMiniSecret(fromHex(data.key!))),
```

Comment:
I believe this is not right.
The miniSecret can not be derived from the private key